sbi/inference/posteriors/sbi_posterior.py

```diff
@@ -287,32 +294,32 @@ def sample(
         self._ensure_single_x(x)
         self._ensure_x_consistent_with_x_o(x)
 
-        with torch.no_grad():
-            if self._sample_with_mcmc:
-                samples = self._sample_posterior_mcmc(
-                    x=x,
-                    num_samples=num_samples,
-                    mcmc_method=self._mcmc_method,
-                    show_progressbar=show_progressbar,
-                    **kwargs,
-                )
-            elif self._alg_family == "snpe":
-                # Rejection sampling.
-                samples, _ = utils.sample_posterior_within_prior(
-                    self.net,
-                    self._prior,
-                    x,
-                    num_samples=num_samples,
-                    show_progressbar=show_progressbar,
-                )
-            else:
-                raise NameError(
-                    "Only SNPE can use rejection sampling. All other"
-                    "methods require MCMC."
-                )
+        if self._sample_with_mcmc:
+            samples = self._sample_posterior_mcmc(
+                x=x,
+                num_samples=num_samples,
+                mcmc_method=self._mcmc_method,
+                show_progressbar=show_progressbar,
+                **kwargs,
+            )
+        elif self._alg_family == "snpe":
+            # Rejection sampling.
+            samples, _ = utils.sample_posterior_within_prior(
+                self.net,
+                self._prior,
+                x,
+                num_samples=num_samples,
+                show_progressbar=show_progressbar,
+            )
+        else:
+            raise NameError(
```

Comment:
The right error is, I think, `ValueError` (the user is passing --indirectly-- `alg_family` as a value which is wrong - incompatible with the rejection setting. 

https://docs.python.org/3/library/exceptions.html#NameError 
https://docs.python.org/3/library/exceptions.html#ValueError